engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/NadelHydrationFieldsBuilder.kt

```diff
@@ -43,8 +51,8 @@ internal object NadelHydrationFieldsBuilder {
         aliasHelper: NadelAliasHelper,
         hydratedField: ExecutableNormalizedField,
         parentNodes: List<JsonNode>,
-    ): List<ExecutableNormalizedField> {
-        val argBatches = NadelBatchHydrationInputBuilder.getInputValueBatches(
+    ): List<CountedBox<ExecutableNormalizedField>> {
+        val boxedArgBatches = NadelBatchHydrationInputBuilder.getInputValueBatches(
```

Comment:
Its the input values that are the key here - a set M parent nodes can lets to N ids to be hydrated